### PR TITLE
Fetch web signup confirmation templates from GraphQL [pr]

### PR DESCRIPTION
**Pull request by aaronschachter**
#### What's this PR do?

Sources signup message content from the GraphQL `webSignupConfirmations` query added in https://github.com/DoSomething/graphql/pull/44, instead of the Gambit Content `GET /campaigns/:id` endpoint.  Per #466, this PR is the last trace of the Gambit Content API in this app.

#### How should this be reviewed?

Verify signups requests are sent for campaign IDs 8226 and 5646.

#### Any background context you want to provide?

Once https://github.com/DoSomething/graphql/pull/47 is deployed to production, we can add a check to prevent sending confirmations for closed campaigns (this may happen when importing signups for campaigns that have ended)

#### Checklist
- [x] Documentation added for new features/changed endpoints.
- [x] Tests added/updated for new features/bug fixes.
- [ ] Tested on staging.

### CODE REVIEW

File: lib/helpers/campaign.js [hunk 1]
```diff
@@ -1,14 +1,38 @@
...
-const gambitContent = require('../gambit-content');
+const graphql = require('../graphql');
+const logger = require('../logger');
+const cache = require('./cache').webSignupConfirmations;
+
+/**
+ * @return {Promise}
+ */
+async function fetchWebSignupConfirmations() {
+  return cache.set(await graphql.fetchWebSignupConfirmations());
```
Comment (rapala61) [inline lib/helpers/campaign.js]: What do you think of creating a test for the scenario where `graphql.fetchWebSignupConfirmations()` fails ? I'm not sure what the caching behavior would be if GraphQL is unavailable.

File: lib/helpers/campaign.js [hunk 2]
```diff
@@ -1,14 +1,38 @@
...
+}
+
+/**
```
Comment (rapala61) [inline lib/helpers/campaign.js]: Just for consistency, I think we could add the descriptions to these function comments?

File: lib/middleware/messages/signup/web-signup-confirmation-get.js [hunk 3]
```diff
@@ -21,19 +16,19 @@ module.exports = function getCampaign() {
...
+      //  This sanity check is useful if importing signups from campaigns that have ended.
+      if (helpers.campaign.isClosedCampaign(template.campaign)) {
+        return helpers.sendErrorResponse(res, new UnprocessableEntityError('Campaign has ended.'));
       }
 
-      helpers.request.setOutboundMessageText(req, webSignup.text);
-      helpers.request.setOutboundMessageTemplate(req, webSignup.template);
-      helpers.request.setTopic(req, webSignup.topic);
+      helpers.request.setOutboundMessageText(req, template.text);
+      helpers.request.setOutboundMessageTemplate(req, 'webSignupConfirmation');
```
Comment (rapala61) [inline lib/middleware/messages/signup/web-signup-confirmation-get.js]: We could move the string `webSignupConfirmation` to a config so we can also use it in tests without hardcoding them.

File: app/routes/messages/signup.js [hunk 4]
```diff
@@ -14,16 +14,16 @@ const getUserMiddleware = require('../../../lib/middleware/messages/user-get');
 const validateOutboundMessageMiddleware = require('../../../lib/middleware/messages/message-outbound-validate');
 const getConversationMiddleware = require('../../../lib/middleware/messages/conversation-get');
 const createConversationMiddleware = require('../../../lib/middleware/messages/conversation-create');
-const getCampaignMiddleware = require('../../../lib/middleware/messages/signup/campaign-get');
+const getWebSignupConfirmationMiddleware = require('../../../lib/middleware/messages/signup/web-signup-confirmation-get');
```
Comment (rapala61) [inline app/routes/messages/signup.js]: This new naming makes so much more sense! 💯

File: test/unit/lib/middleware/messages/message-outbound-create.test.js [hunk 5]
```diff
@@ -64,16 +64,45 @@ test('createOutboundMessage calls next if req.outboundMessage exists', async (t)
...
+  t.context.req.outboundMessageTemplate = stubs.getRandomWord();
   const middleware = createOutboundMessage(configStub);
 
   // test
   await middleware(t.context.req, t.context.res, next);
-  t.context.req.conversation.createAndSetLastOutboundMessage.should.have.been.called;
+  t.context.req.conversation.createAndSetLastOutboundMessage.should.have.been.calledWith(
+    configStub.messageDirection,
+    t.context.req.outboundMessageText,
+    t.context.req.outboundMessageTemplate,
```
Comment (rapala61) [inline test/unit/lib/middleware/messages/message-outbound-create.test.js]: A nit. I would suggest you save the value of `stubs.getRandomWord()` and then compare here. The way the test is comparing against `t.context.req.outboundMessageTemplate` could potentially hide a mutation bug.

**Comment by aaronschachter** [comment]
I've deployed https://github.com/DoSomething/gambit-slack/pull/45 that allows testing signup messages via a new `signup` Slack command:

screen shot 2019-01-28 at 4 52 37 pm

**Pull request merged**